pkg/module/sync/runner.go

```diff
@@ -110,24 +110,28 @@ func (r *Runner) updateManifests(ctx context.Context, kyma *v1beta2.Kyma,
 		return commonerrs.ErrTypeAssert
 	}
 
-	if err := r.doUpdateWithStrategy(ctx, kyma.Labels[shared.ManagedBy], module.Enabled,
-		manifestObj); err != nil {
-		return err
+	if NeedToUpdate(kyma.Status, manifestObj) {
+		if err = r.DoUpdateWithStrategy(ctx, kyma.Labels[shared.ManagedBy], module.Enabled, manifestObj); err != nil {
+			return err
+		}
 	}
 	module.Manifest = manifestObj
+
 	return nil
 }
 
-func (r *Runner) doUpdateWithStrategy(ctx context.Context, owner string, isEnabledModule bool,
+func (r *Runner) DoUpdateWithStrategy(ctx context.Context, owner string, isEnabledModule bool,
 	manifestObj *v1beta2.Manifest,
 ) error {
 	if isEnabledModule {
 		return r.patchManifest(ctx, owner, manifestObj)
 	}
+
 	// For disabled module, the manifest CR is under deleting, in this case, we only update the spec when it's still not deleted.
-	if err := r.updateAvailableManifestSpec(ctx, manifestObj); err != nil && !util.IsNotFound(err) {
+	if err := r.updateAvailableManifestSpec(ctx, manifestObj); client.IgnoreNotFound(err) != nil {
```

Comment:
do you still need this logic? the `manifestObj` should contains disabled module, right?

Reply:
you mean the whole updateAvailableManifestSpec? yes we still need this if the module is being disabled

Reply:
why? if `manifestObj` contains all latest content, you can do the `patchManifest` to disabled module also.

Reply:
the point is we need to ignore the error returned by `patchManifest` if manifest CR is not exists any more.